### PR TITLE
[REVIEW] cuIO reader benchmarks for CSV, ORC and Parquet

### PR DESCRIPTION
**Pull request by devavret**
Small PR to add reader benchmarks for the readers for which the corresponding writer benchmarks were implemented in #5366.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by devavret** [comment]
Results as requested by @ayushdg
<details><summary> Parquet
</summary>
<p>

```
------------------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                                      Time             CPU   Iterations UserCounters...
------------------------------------------------------------------------------------------------------------------------------------------------
ParquetRead<bool>/Boolean_UNCOMPRESSED/536870912/64/0/manual_time                           42.4 ms         42.2 ms           16 bytes_per_second=11.7927G/s
ParquetRead<int8_t>/Byte_UNCOMPRESSED/536870912/64/0/manual_time                            98.3 ms         97.1 ms            7 bytes_per_second=5.08407G/s
ParquetRead<uint8_t>/Ubyte_UNCOMPRESSED/536870912/64/0/manual_time                          98.5 ms         97.2 ms            7 bytes_per_second=5.0783G/s
ParquetRead<int16_t>/Short_UNCOMPRESSED/536870912/64/0/manual_time                          70.1 ms         69.4 ms           10 bytes_per_second=7.13212G/s
ParquetRead<uint16_t>/Ushort_UNCOMPRESSED/536870912/64/0/manual_time                        69.2 ms         68.7 ms           10 bytes_per_second=7.22569G/s
ParquetRead<int32_t>/Int_UNCOMPRESSED/536870912/64/0/manual_time                            73.5 ms         73.5 ms           10 bytes_per_second=6.80651G/s
ParquetRead<uint32_t>/Uint_UNCOMPRESSED/536870912/64/0/manual_time                          73.0 ms         73.0 ms           10 bytes_per_second=6.85299G/s
ParquetRead<int64_t>/Long_UNCOMPRESSED/536870912/64/0/manual_time                           68.7 ms         68.7 ms           10 bytes_per_second=7.2759G/s
ParquetRead<uint64_t>/Ulong_UNCOMPRESSED/536870912/64/0/manual_time                         69.0 ms         69.0 ms           10 bytes_per_second=7.24433G/s
ParquetRead<float>/Float_UNCOMPRESSED/536870912/64/0/manual_time                            73.1 ms         73.1 ms           10 bytes_per_second=6.84035G/s
ParquetRead<double>/Double_UNCOMPRESSED/536870912/64/0/manual_time                          68.5 ms         68.5 ms           10 bytes_per_second=7.29718G/s
ParquetRead<std::string>/String_UNCOMPRESSED/536870912/64/0/manual_time                     33.7 ms         33.6 ms           20 bytes_per_second=14.8261G/s
ParquetRead<cudf::timestamp_D>/Timestamp_days_UNCOMPRESSED/536870912/64/0/manual_time       44.8 ms         44.6 ms           15 bytes_per_second=11.1657G/s
ParquetRead<cudf::timestamp_s>/Timestamp_sec_UNCOMPRESSED/536870912/64/0/manual_time        68.8 ms         68.8 ms           10 bytes_per_second=7.26746G/s
ParquetRead<cudf::timestamp_ms>/Timestamp_ms_UNCOMPRESSED/536870912/64/0/manual_time        68.6 ms         68.7 ms           10 bytes_per_second=7.28497G/s
ParquetRead<cudf::timestamp_us>/Timestamp_us_UNCOMPRESSED/536870912/64/0/manual_time        68.9 ms         68.9 ms           10 bytes_per_second=7.25637G/s
ParquetRead<cudf::timestamp_ns>/Timestamp_ns_UNCOMPRESSED/536870912/64/0/manual_time        68.7 ms         68.8 ms           10 bytes_per_second=7.27393G/s
ParquetRead<bool>/Boolean_USE_SNAPPY/536870912/64/2/manual_time                             42.5 ms         42.2 ms           16 bytes_per_second=11.7667G/s
ParquetRead<int8_t>/Byte_USE_SNAPPY/536870912/64/2/manual_time                               110 ms          108 ms            6 bytes_per_second=4.56391G/s
ParquetRead<uint8_t>/Ubyte_USE_SNAPPY/536870912/64/2/manual_time                             107 ms          107 ms            6 bytes_per_second=4.66183G/s
ParquetRead<int16_t>/Short_USE_SNAPPY/536870912/64/2/manual_time                            80.4 ms         79.7 ms            9 bytes_per_second=6.21558G/s
ParquetRead<uint16_t>/Ushort_USE_SNAPPY/536870912/64/2/manual_time                          80.1 ms         79.1 ms            9 bytes_per_second=6.2441G/s
ParquetRead<int32_t>/Int_USE_SNAPPY/536870912/64/2/manual_time                              92.1 ms         91.7 ms            8 bytes_per_second=5.42966G/s
ParquetRead<uint32_t>/Uint_USE_SNAPPY/536870912/64/2/manual_time                            91.4 ms         91.4 ms            8 bytes_per_second=5.473G/s
ParquetRead<int64_t>/Long_USE_SNAPPY/536870912/64/2/manual_time                              116 ms          114 ms            6 bytes_per_second=4.32426G/s
ParquetRead<uint64_t>/Ulong_USE_SNAPPY/536870912/64/2/manual_time                            111 ms          111 ms            6 bytes_per_second=4.49415G/s
ParquetRead<float>/Float_USE_SNAPPY/536870912/64/2/manual_time                              91.9 ms         91.5 ms            8 bytes_per_second=5.441G/s
ParquetRead<double>/Double_USE_SNAPPY/536870912/64/2/manual_time                            70.4 ms         70.4 ms           10 bytes_per_second=7.10192G/s
ParquetRead<std::string>/String_USE_SNAPPY/536870912/64/2/manual_time                       35.8 ms         35.7 ms           19 bytes_per_second=13.9715G/s
ParquetRead<cudf::timestamp_D>/Timestamp_days_USE_SNAPPY/536870912/64/2/manual_time         52.8 ms         52.1 ms           13 bytes_per_second=9.46236G/s
ParquetRead<cudf::timestamp_s>/Timestamp_sec_USE_SNAPPY/536870912/64/2/manual_time           130 ms          129 ms            5 bytes_per_second=3.84626G/s
ParquetRead<cudf::timestamp_ms>/Timestamp_ms_USE_SNAPPY/536870912/64/2/manual_time           127 ms          127 ms            5 bytes_per_second=3.93744G/s
ParquetRead<cudf::timestamp_us>/Timestamp_us_USE_SNAPPY/536870912/64/2/manual_time          87.3 ms         87.3 ms            8 bytes_per_second=5.72788G/s
ParquetRead<cudf::timestamp_ns>/Timestamp_ns_USE_SNAPPY/536870912/64/2/manual_time          87.3 ms         87.3 ms            8 bytes_per_second=5.72872G/s
```

</p>
</details>

<details><summary> ORC
</summary>
<p>

```
---------------------------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                                   Time             CPU   Iterations UserCounters...
---------------------------------------------------------------------------------------------------------------------------------------------
OrcWrite<bool>/Boolean_UNCOMPRESSED/536870912/64/0/manual_time                            111 ms          111 ms            6 bytes_per_second=4.50348G/s
OrcWrite<int8_t>/Byte_UNCOMPRESSED/536870912/64/0/manual_time                             167 ms          167 ms            4 bytes_per_second=2.99952G/s
OrcWrite<uint8_t>/Ubyte_UNCOMPRESSED/536870912/64/0/manual_time                         0.000 ms        0.000 ms            0
OrcWrite<int16_t>/Short_UNCOMPRESSED/536870912/64/0/manual_time                           123 ms          123 ms            6 bytes_per_second=4.0729G/s
OrcWrite<uint16_t>/Ushort_UNCOMPRESSED/536870912/64/0/manual_time                       0.000 ms        0.000 ms            0
OrcWrite<int32_t>/Int_UNCOMPRESSED/536870912/64/0/manual_time                            79.2 ms         79.2 ms            9 bytes_per_second=6.31254G/s
OrcWrite<uint32_t>/Uint_UNCOMPRESSED/536870912/64/0/manual_time                         0.000 ms        0.000 ms            0
OrcWrite<int64_t>/Long_UNCOMPRESSED/536870912/64/0/manual_time                           59.0 ms         59.0 ms           12 bytes_per_second=8.47755G/s
OrcWrite<uint64_t>/Ulong_UNCOMPRESSED/536870912/64/0/manual_time                        0.000 ms        0.000 ms            0
OrcWrite<float>/Float_UNCOMPRESSED/536870912/64/0/manual_time                            92.8 ms         92.4 ms            8 bytes_per_second=5.38767G/s
OrcWrite<double>/Double_UNCOMPRESSED/536870912/64/0/manual_time                          81.3 ms         81.3 ms            9 bytes_per_second=6.15124G/s
OrcWrite<std::string>/String_UNCOMPRESSED/536870912/64/0/manual_time                     48.3 ms         48.3 ms           14 bytes_per_second=10.3585G/s
OrcWrite<cudf::timestamp_D>/Timestamp_days_UNCOMPRESSED/536870912/64/0/manual_time       66.5 ms         66.6 ms           10 bytes_per_second=7.51487G/s
OrcWrite<cudf::timestamp_s>/Timestamp_sec_UNCOMPRESSED/536870912/64/0/manual_time        72.3 ms         72.4 ms            9 bytes_per_second=6.91341G/s
OrcWrite<cudf::timestamp_ms>/Timestamp_ms_UNCOMPRESSED/536870912/64/0/manual_time        72.2 ms         72.2 ms           10 bytes_per_second=6.92692G/s
OrcWrite<cudf::timestamp_us>/Timestamp_us_UNCOMPRESSED/536870912/64/0/manual_time        80.2 ms         80.2 ms            9 bytes_per_second=6.23709G/s
OrcWrite<cudf::timestamp_ns>/Timestamp_ns_UNCOMPRESSED/536870912/64/0/manual_time        88.2 ms         88.2 ms            8 bytes_per_second=5.67159G/s
OrcWrite<bool>/Boolean_USE_SNAPPY/536870912/64/2/manual_time                              109 ms          109 ms            6 bytes_per_second=4.59017G/s
OrcWrite<int8_t>/Byte_USE_SNAPPY/536870912/64/2/manual_time                               178 ms          178 ms            4 bytes_per_second=2.80494G/s
OrcWrite<uint8_t>/Ubyte_USE_SNAPPY/536870912/64/2/manual_time                           0.000 ms        0.000 ms            0
OrcWrite<int16_t>/Short_USE_SNAPPY/536870912/64/2/manual_time                             173 ms          173 ms            4 bytes_per_second=2.8862G/s
OrcWrite<uint16_t>/Ushort_USE_SNAPPY/536870912/64/2/manual_time                         0.000 ms        0.000 ms            0
OrcWrite<int32_t>/Int_USE_SNAPPY/536870912/64/2/manual_time                              94.7 ms         94.7 ms            7 bytes_per_second=5.28121G/s
OrcWrite<uint32_t>/Uint_USE_SNAPPY/536870912/64/2/manual_time                           0.000 ms        0.000 ms            0
OrcWrite<int64_t>/Long_USE_SNAPPY/536870912/64/2/manual_time                             66.5 ms         66.6 ms           10 bytes_per_second=7.51391G/s
OrcWrite<uint64_t>/Ulong_USE_SNAPPY/536870912/64/2/manual_time                          0.000 ms        0.000 ms            0
OrcWrite<float>/Float_USE_SNAPPY/536870912/64/2/manual_time                              94.4 ms         94.4 ms            7 bytes_per_second=5.29848G/s
OrcWrite<double>/Double_USE_SNAPPY/536870912/64/2/manual_time                            83.7 ms         83.7 ms            8 bytes_per_second=5.97698G/s
OrcWrite<std::string>/String_USE_SNAPPY/536870912/64/2/manual_time                       56.6 ms         56.5 ms           12 bytes_per_second=8.83315G/s
OrcWrite<cudf::timestamp_D>/Timestamp_days_USE_SNAPPY/536870912/64/2/manual_time         71.5 ms         71.5 ms           10 bytes_per_second=6.99295G/s
OrcWrite<cudf::timestamp_s>/Timestamp_sec_USE_SNAPPY/536870912/64/2/manual_time          73.6 ms         73.6 ms            9 bytes_per_second=6.79385G/s
OrcWrite<cudf::timestamp_ms>/Timestamp_ms_USE_SNAPPY/536870912/64/2/manual_time          74.2 ms         74.2 ms            9 bytes_per_second=6.73931G/s
OrcWrite<cudf::timestamp_us>/Timestamp_us_USE_SNAPPY/536870912/64/2/manual_time          88.1 ms         88.1 ms            8 bytes_per_second=5.67799G/s
OrcWrite<cudf::timestamp_ns>/Timestamp_ns_USE_SNAPPY/536870912/64/2/manual_time           100 ms          100 ms            7 bytes_per_second=4.99862G/s
```

</p>
</details>

**Comment by vuule** [comment]
@devavret Why are there no CSV results?

**Comment by devavret** [comment]
> @devavret Why are there no CSV results?

@ayushdg wanted a comparison between parquet and ORC.

**Comment by vuule** [comment]
> > @devavret Why are there no CSV results?
> 
> @ayushdg wanted a comparison between parquet and ORC.

My bad, could not find the requesting comment so I had no context.

File: cpp/benchmarks/io/orc/orc_reader_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+constexpr int64_t data_size = 512 << 20;  // 512 MB
+
+namespace cudf_io = cudf::io;
+
+template <typename T>
+class OrcWrite : public cudf::benchmark {
+};
+
+template <typename T>
+void ORC_write(benchmark::State& state)
```
Comment (harrism) [inline cpp/benchmarks/io/orc/orc_reader_benchmark.cpp:37]: Should this be ORC_read?

```suggestion
void ORC_read(benchmark::State& state)
```

**Pull request merged**